test/persistence-hooks.suite.js

```diff
@@ -3257,13 +3256,8 @@ module.exports = function(dataSource, should, connectorCapabilities) {
                 id: existingInstance.id,
                 name: 'updated name',
               },
+              isNewInstance: false,
             });
-            // For non-atomic implementation of upsertWithWhere on update, it calls
-            // updateAttributes. loaded hook of updateAttributes does not provide
-            // isNewInstance.
-            if (dataSource.connector.upsertWithWhere) {
-              expectedContext.isNewInstance = false;
-            }
```

Comment:
This should have been caught by running `npm test` locally. 

@Amir-61 could you please fix tests of the memory connector to correctly build a connector with and without an atomic implementation of `upsertWithWhere`?

See here:

https://github.com/strongloop/loopback-datasource-juggler/blob/5bf18b07280e22d713e722b8e23f8aee14c6587e/test/memory.test.js#L890-L908

IMO, the correct solution is:
 - The `memory` connector provides optimized/atomic `upsertWithWhere`, therefore no changes are needed in `describe('Optimized connector')`
 - Inside `describe('Unoptimized connector')`, we need to set `upsertWithWhere=false` to trigger the other branch inside `DAO.upsertWithWhere`.